File: OneToOneCallSample/app/src/test/java/com/stringee/apptoappcallsample/stringee/common/CallPermissionsTest.java
package com.stringee.apptoappcallsample.stringee.common;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.robolectric.Shadows.shadowOf;

import android.app.NotificationManager;
import android.content.Context;

import androidx.test.core.app.ApplicationProvider;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

@RunWith(RobolectricTestRunner.class)
@Config(sdk = 32)
public class CallPermissionsTest {
    private final Context context = ApplicationProvider.getApplicationContext();

    @Test
    public void notificationAccessIncludesTheSystemNotificationToggle() {
        NotificationManager manager = context.getSystemService(NotificationManager.class);
        shadowOf(manager).setNotificationsEnabled(true);
        assertTrue(CallPermissions.canPostNotifications(context));

        shadowOf(manager).setNotificationsEnabled(false);
        assertFalse(CallPermissions.canPostNotifications(context));
    }
}

File: Kotlin-OneToOneCallSample/app/src/test/java/com/stringee/kotlin_onetoonecallsample/stringee/common/CallPermissionsTest.kt
package com.stringee.kotlin_onetoonecallsample.stringee.common

import android.app.NotificationManager
import android.content.Context
import androidx.test.core.app.ApplicationProvider
import com.stringee.kotlin_onetoonecallsample.stringee.common.CallPermissions.canPostNotifications
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.Shadows
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [32])
class CallPermissionsTest {
    private val context: Context = ApplicationProvider.getApplicationContext<Context>()

    @Test
    fun notificationAccessIncludesTheSystemNotificationToggle() {
        val manager =
            context.getSystemService<NotificationManager?>(NotificationManager::class.java)
        Shadows.shadowOf(manager).setNotificationsEnabled(true)
        Assert.assertTrue(canPostNotifications(context))

        Shadows.shadowOf(manager).setNotificationsEnabled(false)
        Assert.assertFalse(canPostNotifications(context))
    }
}
